packages/trie/src/node/branch.ts

```diff
@@ -1,14 +1,12 @@
 import { RLP } from '@ethereumjs/rlp'
 
-import type { EmbeddedNode } from '../types.js'
-
 export class BranchNode {
-  _branches: (EmbeddedNode | null)[]
-  _value: Uint8Array | null
+  _branches: (Uint8Array | Uint8Array[] | null)[]
+  _value: Uint8Array
 
   constructor() {
     this._branches = new Array(16).fill(null)
-    this._value = null
+    this._value = new Uint8Array(0)
```

Comment:
This will not work, now all branch nodes have value 0. This is not possible, for instance for the unhashed trie where we have these key/values:

`dog` -> 1
`done` -> 2

If you would index this on `do` (or `d`) it would return value 0, where it should be `null`. This will likely not mess up the hash of the tries, but it will mess up people using the tries (likely the unhashed versions) as key/value store. They would expect empty values to return `null` and not empty Uint8Arrays.